HeroVault/HeroVault/Features/Home/HomeScreen/HomeScreenViewModel.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  HomeScreenViewModel.swift
+//  HeroVault
+//
+//  Created by Gia Catano on 2024/03/30.
+//
+
+import Foundation
+
+protocol ViewModelProtocol: AnyObject {
+    func reloadView()
+}
+
+class HomeScreenViewModel {
+    
+    // MARK: Variables
+    
+    var marvelData = [MarvelData]()
+    var filteredMarvelData = [MarvelData]()
+    var marvelDataType: EntityType
+    var error: Error?
+    var isSearching: Bool
+    var hideNoResultsText: Bool
+    
+    private let homeScreenRepository: HomeScreenRepositoryType?
+    private weak var delegate: ViewModelProtocol?
+    
+    init(homeScreenRepository: HomeScreenRepositoryType, delegate: ViewModelProtocol) {
+        self.homeScreenRepository = homeScreenRepository
+        self.delegate = delegate
+        self.marvelDataType = .character
+        isSearching = false
```

Comment:
you can set these default vaules where you are declaring the properties

Reply:
yeah you can!
My opinion is different from Simekani's though. I personally like to instantiate stuff in the init (only if there is an init already that is being used to instantiate other stuff with constructor DI). 
If there wasn't an init, i do what Simekani suggests, but When there is an init, i do what you have done. Up to you, think it's more of a coding preference.